zuul-core/src/main/java/com/netflix/zuul/message/http/Cookies.java

```diff
@@ -16,25 +16,56 @@
 
 package com.netflix.zuul.message.http;
 
+import com.netflix.config.CachedDynamicBooleanProperty;
+import com.netflix.zuul.message.Headers;
 import io.netty.handler.codec.http.Cookie;
-
+import io.netty.handler.codec.http.DefaultCookie;
+import io.netty.handler.codec.http.cookie.ServerCookieDecoder;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * User: Mike Smith
- * Date: 6/18/15
- * Time: 12:04 AM
+ * User: Mike Smith Date: 6/18/15 Time: 12:04 AM
  */
-public class Cookies
-{
+public class Cookies {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Cookies.class);
+
+    private static final CachedDynamicBooleanProperty CLEAN_COOKIES = new CachedDynamicBooleanProperty(
+            "zuul.HttpRequestMessage.cookies.clean", false);
+
     private Map<String, List<Cookie>> map = new HashMap<>();
     private List<Cookie> all = new ArrayList<>();
 
-    public void add(Cookie cookie)
-    {
+    public static Cookies fromHeaders(Headers headers) {
+        Cookies cookies = new Cookies();
+        for (String aCookieHeader : headers.get(HttpHeaderNames.COOKIE)) {
+            try {
+                if (CLEAN_COOKIES.get()) {
+                    aCookieHeader = CookiesSanitizer.cleanCookieHeader(aCookieHeader);
+                }
+                List<io.netty.handler.codec.http.cookie.Cookie> decodedCookies =
```

Comment:
While I understand this isn't directly part of your change, it'd be nice if we could take this opportunity to encapsulate this logic to `Cookies` instead.
Keeping the deprecated `Cookie` local to that class, consider adding a utility method that can do the mapping given a collection of `io.netty.handler.codec.http.cookie.Cookie`?